python/ray/train/batch_predictor.py

```diff
@@ -97,6 +97,9 @@ def predict(
         predictor_cls = self.predictor_cls
         checkpoint_ref = self.checkpoint_ref
         predictor_kwargs = self.predictor_kwargs
+        # Automatic set use_gpu in predictor constructor if user provided
+        # explicit GPU resources
+        predictor_kwargs["use_gpu"] = num_gpus_per_worker > 0
```

Comment:
These are kwargs for predict, but use_gpu should be set in `from_checkpoint` right?

Reply:
Also, I think we should automatically set `use_gpu` only if it was not explicitly disabled by the user.

```suggestion
        predictor_kwargs["use_gpu"] = predictor_kwargs.get("user_gpu", num_gpus_per_worker > 0)
```

Reply:
Can we also do the same automatic setting in `PredictorDeployment` for the online serving case?

Reply:
I think there're two sets of kwargs in this file: `predictor_kwargs` is kwargs we used to initialize predictor which we want to have `use_gpu` populated as expected; There's also `predict_kwargs` that is passed into each `predict()` call

Reply:
@amogkam so i checked `PredictorDeployment ` and found its interface quite different from `BatchPredictor`. 

In `BatchPredictor` there's a predict function with signature of 

```
    def predict(
        self,
        data: Union[ray.data.Dataset, ray.data.DatasetPipeline],
        *,
        batch_size: int = 4096,
        min_scoring_workers: int = 1,
        max_scoring_workers: Optional[int] = None,
        num_cpus_per_worker: int = 1,
        num_gpus_per_worker: int = 0,
        ray_remote_args: Optional[Dict[str, Any]] = None,
        **predict_kwargs,
    ) -> ray.data.Dataset:
```

While `PredictorDeployment` directly takes the input that corresponds for each request:

```
    async def predict(self, inp):
        """Perform inference directly without HTTP."""
        return await self.predict_impl(inp)
```

Reply:
Ah sorry I misread the name! You are right it should be in `predictor_kwargs`.

Should we use Python `inspect` to only add it to predictor_kwargs if the arg exists in `from_checkpoint`?

Also regarding `PredictorDeployment`, the constructor takes in `predictor_from_checkpoint_kwargs` and we would want to add it there. @simon-mo is there a way to tell inside the Deployment if gpus have been enabled?